lib/intrinsicclass.ts

```diff
@@ -1,139 +0,0 @@
-    (new (
-      epochNanoseconds: JSBI,
-      timeZone: Temporal.TimeZoneProtocol | string,
-      calendar?: Temporal.CalendarProtocol | string
-    ) => Temporal.ZonedDateTime) & {
-      prototype: typeof Temporal.ZonedDateTime.prototype;
-      from: typeof Temporal.ZonedDateTime.from;
-      compare: typeof Temporal.ZonedDateTime.compare;
-    };
-};
-type TemporalIntrinsicRegistrations = {
-  [key in keyof TemporalIntrinsics as `Temporal.${key}`]: TemporalIntrinsics[key];
-};
-type TemporalIntrinsicPrototypeRegistrations = {
-  [key in keyof TemporalIntrinsics as `Temporal.${key}.prototype`]: TemporalIntrinsics[key]['prototype'];
-};
-type TemporalIntrinsicRegisteredKeys = {
-  [key in keyof TemporalIntrinsicRegistrations as `%${key}%`]: TemporalIntrinsicRegistrations[key];
-};
-type TemporalIntrinsicPrototypeRegisteredKeys = {
-  [key in keyof TemporalIntrinsicPrototypeRegistrations as `%${key}%`]: TemporalIntrinsicPrototypeRegistrations[key];
-};
-
-interface StandaloneIntrinsics {
-  'Temporal.Calendar.from': typeof Temporal.Calendar.from;
-}
-type RegisteredStandaloneIntrinsics = { [key in keyof StandaloneIntrinsics as `%${key}%`]: StandaloneIntrinsics[key] };
-const INTRINSICS: Partial<TemporalIntrinsicRegisteredKeys> &
-  Partial<TemporalIntrinsicPrototypeRegisteredKeys> &
-  Partial<RegisteredStandaloneIntrinsics> = {};
-
-type customFormatFunction<T> = (
-  this: T,
-  depth: number,
-  options: { stylize: (value: unknown, type: 'number' | 'special') => string }
-) => string;
-const customUtilInspectFormatters: Partial<{
-  [key in keyof TemporalIntrinsicRegistrations]: customFormatFunction<
-    InstanceType<TemporalIntrinsicRegistrations[key]>
-  >;
-}> = {
-  ['Temporal.Duration'](depth, options) {
-    const descr = options.stylize(`${this[Symbol.toStringTag]} <${this}>`, 'special');
-    if (depth < 1) return descr;
-    const entries = [];
-    for (const prop of [
-      'years',
-      'months',
-      'weeks',
-      'days',
-      'hours',
-      'minutes',
-      'seconds',
-      'milliseconds',
-      'microseconds',
-      'nanoseconds'
-    ] as const) {
-      if (this[prop] !== 0) entries.push(`  ${prop}: ${options.stylize(this[prop], 'number')}`);
-    }
-    return descr + ' {\n' + entries.join(',\n') + '\n}';
-  }
-};
-
-type InspectFormatterOptions = { stylize: (str: string, styleType: string) => string };
-function defaultUtilInspectFormatter(this: any, depth: number, options: InspectFormatterOptions) {
-  return options.stylize(`${this[Symbol.toStringTag]} <${this}>`, 'special');
-}
-
-export function MakeIntrinsicClass(
-  Class: TemporalIntrinsicRegistrations[typeof name],
-  name: keyof TemporalIntrinsicRegistrations
-) {
-  Object.defineProperty(Class.prototype, Symbol.toStringTag, {
-    value: name,
-    writable: false,
-    enumerable: false,
-    configurable: true
-  });
-  if (DEBUG) {
-    Object.defineProperty(Class.prototype, Symbol.for('nodejs.util.inspect.custom'), {
```

Comment:
Is this functionality (which AFAIK provides a way for Node to show friendlier output in the debugger) moved somewhere else?  If not, then it probably should be, unless for some reason there's a better way to handle this.

Reply:
@ptomato is this currently in use somewhere (as you added this)? Seems like a few other libraries (e.g. [Moment.js](https://github.com/moment/moment/blob/e96809208c9d1b1bbe22d605e76985770024de42/moment.js#L4914)) use this, but doesn't seem to be super common. If it's needed I would create a `debug.ts`, add the method there and attach it to the classes explicitly.

Reply:
This is mainly useful when debugging or testing things out interactively. I care mostly that it is present in proposal-temporal, I think it's probably useful here as well, but I'd leave it up to the people who maintain this polyfill more actively than me.